### ISSUE TITLE
Methods for Efficiently Predicting on a Test set (without retraining)

### ISSUE THREAD

**Issue by maximilianreimer** [unit8co/darts#1275]
I am using the TFTModel. After training (and validating) using the `fit` method, I would like to predict all data points in the train, test and validation set using the already trained model. 

Currently, there are only the methods: 

* `historical_forcast`: supports predicting for multiple time steps (with corresponding look backs) but just one time series
* `predict`:  supports predicting for multiple time series but just for n next time steps. 

What I am looking for is a method like `historical_forcast` but where `series`, `past_covariates`,  and `future_covariates` are supported for being predicted without retraining. My best attempt so far is to run the following code block on an already trained model:  

```python
predictions = []
for s, past_cov, future_cov in zip(series, past_covariates, future_covariates): 
      predictions.append(model.historical_forecasts(
              s,
              past_covariates=past_cov,
              future_covariates=future_cov,
              retrain=False,
              start=model.input_chunk_length,
              verbose=True
          ))
```

Where `series`, `past_covariates`,  and `future_covariates` are lists of target time series and covariates respectively, each consisting of the concatenated train, val and test series which I split afterwards again to ensure the availability of the past values needed for predicting at the beginning of test and val. 

My objection / question about this: is there a more efficient way to do this through better batching with the current interface, our would I have to call the torch model my self?

**Comment by dennisbader** [comment]
Hi @maximilianreimer, you are right, it's currently only possible to use hostorical forecasts using single series.

It is technically possible (with some extra effort) to build a custom inference dataset and use `model.predict_from_dataset()`. You could inherit from Darts `MixedCovariatesSequentialDataset` and overwrite how the inference batches should be created. This would include how to extract subsamples from your series similar to historical forecasts.

**Comment by maximilianreimer** [comment]
Hi @dennisbader,

I essentially started doing this, but with the predict method, by chunking the into overlapping series I want to predict. One issue is now, that there seems to be a bug with the predict method. 

```python

chunks = ... # List of TimeSeries of each of length model.input_chunk_length 
batch_size = 16
preditcted_chunks = model.predict(model.output_chunk_length, chunks, batch_size)

assert len(preditcted_chunks) == batch_size # successful
assert len(preditcted_chunks) == len(chunks) # fails
 ```
For some reason, only the first batch is  predicted. I am using u8darts 0.21.0..  Any ideas why?

**Comment by dennisbader** [comment]
Okay, that should also work.

Could you provide a minimum code snippet to reproduce this issue? You can use Darts timeseries generation to create some dummy data if you want.

**Comment by maximilianreimer** [comment]
Silly mistake! I set 
```
TFTModel(
...
   pl_trainer_kwargs={
                    "fast_dev_run": True,
   }
...
)
```

**Comment by hrzn** [comment]
We're working on adding support for multiple series in `historical_forecasts` here: https://github.com/unit8co/darts/pull/1318
